replayserver/bookkeeping/bookkeeper.py

```diff
@@ -38,26 +33,7 @@ def build(cls, database, config):
             logger.debug(f"Saved replay {game_id}")
             metrics.saved_replays.inc()
             logger.debug(f"Parsing replay {game_id}")
-            ticks = await self.get_ticks(stream)
+            ticks = get_replay_ticks(stream.data.bytes())
```

Comment:
I'd rather this was injected, even if we don't have unit tests for bookkeeper.

Reply:
Why?

Reply:
Because unit testability is nice.

Reply:
You don't need dependency injection in order to write unit tests. I already wrote a basic test for this function

Reply:
I can see a test for `get_replay_ticks`, I'm talking about Bookkeeper. The way it is now, we can't test Bookkeeper itself without testing replay parsing at the same time.

Reply:
The test isn't supposed to verify that the replay parser works correctly, it's supposed to verify that `get_replay_ticks` calls the parser correctly. For example, when testing initially I forgot to call `parser.parse()` after creating the parser. This resulted in there being no errors, but the tick count was always 0. This is the sort of thing the test verifies. 

Really what needs to be tested is that the actual data from `stream.data.bytes()` of a real replay is parsed correctly. This isn't going to happen with a unit test anyways, it will require an integration test. Why would you need to test the bookkeeper without replay parsing, if it's never going to actually run like that? What would be the benefit of that test? You would just be mocking all real work and simply be checking that python code executes in order.

Reply:
I want it mostly out of principle - I tried my best to avoid strong dependencies in code. I could've used the same argument for different classes, but then their behaviour became complex enough to warrant testing..

Reply:
`import this`

Reply:
Sure, let me quote you one: `Special cases aren't special enough to break the rules.`

Reply:
First of all, this is not special case. In fact, you really don't wanna have any of special cases, you want to have code as simple as it's possible.

And dude, you're making "big project" or "small tiny service"? I think, it's small tiny service, nothing else. Why are you overdoing it?

Now you're searching the excuses do to make things to complicated, but forgetting and omitting basics. (https://github.com/FAForever/java-guidelines/wiki/Contribution-Guidelines#you-are-not-your-code)

And how you wrote it now, I've been spend at least 8 hours reading your code, and trying to understand, what it's doing. And still I didn't get it. 6 classes in one file, totally lack of typing, short variables, functions that you have to read 3 mintues to understand, what's doing, you used some other package libraries, than we use on other projects, different type of configs, i can continue and continue... Now you're trying to have some dependency injections? That code is already not maintainable! So now there is only you who understand what's going on on that server, not others.

Like I would like to help here, but I can't work on it, because I don't wanna to read everything from beginning for 8 hours again and try to understand.

Reply:
None of this has anything to do with the PR. If you believe this codebase is unmaintainable or otherwise bad, discuss it on Zulip and see if others share your opinion.